omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/BedDetailsResourceSearchHandler.java

```diff
@@ -44,13 +45,10 @@ public SearchConfig getSearchConfig() {
 	
 	@Override
 	public PageableResult search(RequestContext requestContext) throws ResponseException {
-		BedManagementService bedManagementService = (BedManagementService) Context
-		        .getModuleOpenmrsServices(BedManagementService.class.getName()).get(0);
 		String visitUuid = requestContext.getParameter("visitUuid");
-		BedDetails bedDetails = bedManagementService.getLatestBedDetailsByVisit(visitUuid);
-		AlreadyPaged<BedDetails> alreadyPaged = new AlreadyPaged<BedDetails>(requestContext,
-		        Collections.singletonList(bedDetails), false);
-		return bedDetails == null ? new EmptySearchResult() : alreadyPaged;
+		BedDetails bedDetails = Context.getService(BedManagementService.class).getLatestBedDetailsByVisit(visitUuid);
+		List<BedDetails> ret = bedDetails == null ? new ArrayList<>() : Collections.singletonList(bedDetails);
```

Comment:
Does it make sense to use `Collections.emptyList()` here?

Reply:
Sure, done